src/cfml/system/services/ServerEngineService.cfc

```diff
@@ -0,0 +1,177 @@
+    
+    return installDir;
+  }
+
+
+	/*
+	* Downloads a CF engine endpoint and unzips the "Engine.[WAR|zip] archive into the destination
+	*/
+	function installEngineArchive( required string ID, required string destination, required string cfengine, string installDir ) {
+    	var thisTempDir = tempDir & '/' & createUUID();
+    	var versionInstalled = '0';
+    	 
+		packageService.installPackage( ID=arguments.ID, directory=thisTempDir, save=false );
+				
+		if( packageService.isPackage( thisTempDir ) ) {
+			var boxJSON = packageService.readPackageDescriptor( thisTempDir );
+			versionInstalled =  boxJSON.version;
+		}
+				
+		var theArchive = '';
+		for( var thisFile in directoryList( thisTempDir ) ) {
+			if( listFindNoCase( 'war,zip', listLast( thisFile, '.' ) ) ) {
+				theArchive = thisFile;
+			}
+		}
+		
+		if( theArchive == '' ) {
+			throw( "Package didn't contain a war or zip archive." );
+		}
+				
+    	var installDir = arguments.installDir ?: destination & cfengine & "-" & replace( versionInstalled, '+', '.', 'all' );
+		
+		consoleLogger.info( "Exploding WAR/zip archive...");
+		zip action="unzip" file="#thisFile#" destination="#installDir#" overwrite="true";
+				
+		// Catch this to gracefully handle where the OS or another program 
+		// has the folder locked.
+		try {
+			directoryDelete( thisTempDir, true );
+		} catch( any e ) {
+			consoleLogger.error( '#e.message##CR#The folder is possibly locked by another program.' );
+			logger.error( '#e.message# #e.detail#' , e.stackTrace );
+		}
+		return installDir;
+	}
+  /**
+  * configure web.xml file for Lucee and Railo
+  * @cfengine        lucee or railo
+  * @source          source web.xml
+  * @destination     target web.xml
+  **/
+  public function configureWebXML(required cfengine, required source, destination) {
+  	var webXML = XMLParse(source);
+    var servlets = xmlSearch(webXML,"//:servlet-class[text()='#lcase(cfengine)#.loader.servlet.CFMLServlet']");
+    var initParam = xmlElemnew(webXML,"http://java.sun.com/xml/ns/javaee","init-param");
+    initParam.XmlChildren[1] = xmlElemnew(webXML,"param-name");
+    initParam.XmlChildren[1].XmlText = "#lcase(cfengine)#-web-directory";
+    initParam.XmlChildren[2] = xmlElemnew(webXML,"param-value");
+    initParam.XmlChildren[2].XmlText = "/WEB-INF/#lcase(cfengine)#/{web-context-label}";
+    arrayInsertAt(servlets[1].XmlParent.XmlChildren,4,initParam);
+		var xlt = '<xsl:stylesheet version="1.0" xmlns:xsl="http://www.w3.org/1999/XSL/Transform">
+		<xsl:output method="xml" encoding="utf-8" indent="yes" xslt:indent-amount="2" xmlns:xslt="http://xml.apache.org/xslt" />
+		<xsl:strip-space elements="*"/>
+		<xsl:template match="node() | @*"><xsl:copy><xsl:apply-templates select="node() | @*" /></xsl:copy></xsl:template>
+		</xsl:stylesheet>';
+    fileWrite(destination,toString(XmlTransform(webXML,xlt)));
+    return true;
+  }
+
+  /**
+  * Dynamic completion for cfengine
+  */  
+  function getCFEngineNames() {
```

Comment:
Replace this with ForgeBox search for projects of type "cf-engines" and their versions.